modules/accounts/calendar.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.accounts.calendar;
+
+  calendarOpts = { name, config, ... }: {
+    options = {
```

Comment:
I would have expected a caldav sub-module here with some connection information. I saw that `vdirsyncer-accounts.nix` contains that sort of stuff. So would be nice to move that here.

Similarly the contacts module would benefit from having a carddav sub-module.

Reply:
What would you obtain these respective submodules to contain?

Reply:
the carddav/caldav could contain url, a way to pass on the password, eventually a type like "nextcloud"/"owncloud"/"google" to automatically adapt some parameters like https://github.com/rycee/home-manager/blob/f83c49baa3cf4de627bcfd3c70730478aaa3998a/modules/accounts/email.nix#L87